src/index.ts

```diff
@@ -117,6 +117,19 @@ let resolver: Resolver | undefined
 const digestHashObject = (value: object | null | undefined) =>
   hashObject(value ?? {}).digest('hex')
 
+/**
+ * Checks if a module is a core module
+ * module.isBuiltin is available in Node.js 16.17.0 or later. Once we drop support for older
+ * versions of Node.js, we can use module.isBuiltin instead of this function.
+ */
+function isBuiltin(moduleName: string) {
+  const moduleNameWithoutPrefix = moduleName.startsWith('node:')
+    ? moduleName.slice(5)
+    : moduleName;
+
+  return builtinModules.includes(moduleNameWithoutPrefix);
```

Comment:
i think we can just do 
```suggestion
  return builtinModules.includes(moduleName.replace(/^node:/, ''));
```

Reply:
Good call!